consensus/src/dag/types.rs

```diff
@@ -164,38 +184,86 @@ impl Node {
         &self.metadata
     }
 
-    pub fn parents(&self) -> &[NodeMetadata] {
+    pub fn parents(&self) -> &[NodeCertificate] {
         &self.parents
     }
 
     pub fn author(&self) -> &Author {
         self.metadata.author()
     }
+
+    pub fn sign(&self, signer: &ValidatorSigner) -> Result<Signature, CryptoMaterialError> {
+        let node_digest = NodeDigest::new(self.digest());
+        signer.sign(&node_digest)
+    }
+}
+
+impl TDAGMessage for Node {
+    fn verify(&self, verifier: &ValidatorVerifier) -> anyhow::Result<()> {
+        let current_round = self.metadata().round();
+
+        let prev_round = current_round - 1;
+        // check if the parents' round is the node's round - 1
+        ensure!(
+            self.parents()
+                .iter()
+                .all(|parent| parent.metadata().round() == prev_round),
+            "invalid parent round"
+        );
+
+        ensure!(
```

Comment:
need to handle round 0?

Reply:
it is a special case I am handling in `reliable_broadcast.rs`